src/defaultsettings.cpp

```diff
@@ -196,11 +252,7 @@ void set_default_settings()
 	settings->setDefault("screen_h", "600");
 	settings->setDefault("window_maximized", "false");
 	settings->setDefault("autosave_screensize", "true");
-#ifdef ENABLE_TOUCH
-	settings->setDefault("fullscreen", "true");
-#else
-	settings->setDefault("fullscreen", "false");
-#endif
+	settings->setDefault("fullscreen", bool_to_cstr(has_touch));
```

Comment:
uhh surely not everyone with touch automatically wants fullscreen?

Reply:
usually if you have a touchscreen, the gestures are used (at least Linux + Android), so if the app opens in fullscreen, you can swipe and switch it back to windowed mode (or open shell).

Currently I have opposite problem. After sliding finger to get back to shell from Minetest, it'll turn windowed and not return into fullscreen after selecting it again.

Reply:
@sfan5 what do you think?

Reply:
Feel free to leave it like this.